utils/utils.go

```diff
@@ -0,0 +1,103 @@
+package utils
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	"github.com/go-continuous-fuzz/go-continuous-fuzz/config"
+	"github.com/otiai10/copy"
+)
+
+const HelpText = `Usage: go run main.go [command]
+
+Commands:
+  help      Show this help message and exit.
+
+Environment Variables:
+  FUZZ_NUM_PROCESSES
+          Specifies the number of fuzzing processes to run concurrently.
+          Default: Maximum number of CPU cores available on the machine.
+
+  PROJECT_SRC_PATH    (Required)
+          The Git repository URL of the project to be fuzzed.
+          Formats:
+            - Private: https://oauth2:PAT@github.com/OWNER/REPO.git
+            - Public:  https://github.com/OWNER/REPO.git
+
+  GIT_STORAGE_REPO    (Required)
+          The Git repository where the input corpus is stored.
+          Format: https://oauth2:PAT@github.com/OWNER/STORAGEREPO.git
+
+  FUZZ_TIME
+          Duration (in seconds) for which the fuzzing engine should run.
+          Default: 120 seconds.
+
+  FUZZ_PKG   (Required)
+          The specific Go package within the repository to be fuzzed.
+
+  FUZZ_RESULTS_PATH
+          Path to store fuzzing results, relative to the current working
+	  directory
+          Default: Project root directory
+
+Usage Example:
+  Set the necessary environment variables, then start fuzzing:
+      go run main.go
+
+For more information, please refer to the project documentation.`
+
+// CleanupWorkspace deletes the "out" directory to reset the workspace state.
+// Any errors encountered during removal are logged, but do not stop execution.
+func CleanupWorkspace(logger *slog.Logger) {
+	if err := os.RemoveAll("out"); err != nil {
+		logger.Error("workspace cleanup failed", "error", err)
+	}
+}
+
+// SaveFuzzCorpus copies the generated corpus data for a given package and
+// target to the configured fuzz results directory. If the corpus directory does
+// not exist, it logs an informational message and returns. Any errors during
+// directory creation or copying are logged and cause the process to exit.
+func SaveFuzzCorpus(logger *slog.Logger, cfg *config.Config, pkg,
+	target string) {
+
+	corpusPath := filepath.Join(config.DefaultCorpusDir, pkg, "testdata",
+		"fuzz", target)
```

Comment:
So the current corpus flow is:

1. Initial corpus is downloaded from GitHub (`Config.GitStorageRepo`).
2. Fuzzing is done and inputs are added to the initial corpus.
3. Updated corpus is copied to `Config.DefaultCorpusDir`.

Then on restart the updated corpus is discarded and the initial corpus is redownloaded from GitHub again, thereby discarding any progress made while fuzzing.

For now, I think what we want is to get rid of the GitHub step altogether and start from a corpus directory specified by the user.  When fuzzing is done, that corpus directory will automatically have any new inputs found by the fuzzer, so progress will be saved between restarts.